### DISCUSSION TITLE
Removing Theia Context menu items on conditional basis

### DISCUSSION THREAD

**Discussion by Shulabh-Prakash** [eclipse-theia/theia#14555] [Q&A]
I need to remove some specific context menu items (such as "Select for Compare," "Copy Path," etc.) based on certain conditions related to the URI. Specifically, I want to conditionally hide or remove these items based on the URI that’s being used. Is there a way to achieve this.
Any help or pointers on how to approach this would be greatly appreciated!

**Comment by msujew** [comment]
Hey @Shulabh-Prakash,

you will need to unregister the the associated commands. See here for the command used in the `Copy Path` context menu entry:

https://github.com/eclipse-theia/theia/blob/c8ed4981bef563c4a783e9e2dbcd55587b526748/packages/navigator/src/browser/navigator-contribution.ts#L399-L402

After unregistering them, you can register a new handler for the command, that uses an `isEnabled` function to ensure that the command is only enabled/part of the context menu if the URI satisfies a certain condition. See for an example:

https://github.com/eclipse-theia/theia/blob/c8ed4981bef563c4a783e9e2dbcd55587b526748/packages/navigator/src/browser/navigator-contribution.ts#L234

**Reply by Shulabh-Prakash** [reply]
Hii   @msujew ,
I have a question regarding the process of changing a command’s visibility and enabled state. 
So, if I want to change a command such as one below

![save_path](https://github.com/user-attachments/assets/1925f2da-6c62-4d00-8984-f3d59caa5162)

From my understanding, in order to update the isVisible and isEnabled properties, I would need to unregister the command from its current registration and then re-register it with the desired changes.

However, it seems like this might require duplicating the entire execute function. Is there a more efficient way to achieve this, where I don’t have to duplicate the execute function while updating these properties?

Looking forward to your insights!

**Reply by msujew** [reply]
I guess you could use the `getAllHandlers` method of the `CommandRegistry` and override the `isEnabled` function field directly. However, I'm not sure how well this works in combination with the underlying `UriAwareCommandHandler`.
